tests/controller/controller_test.go

```diff
@@ -0,0 +1,628 @@
+package controller_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/awslabs/aws-s3-csi-driver/pkg/podmounter/mppod"
+)
+
+var _ = Describe("Mountpoint Controller", func() {
+	Context("Static Provisioning", func() {
```

Comment:
Are there any tests for volumes created with say hostpath/emptydir to verify they're not scheduled?

Reply:
There are some test cases for checking PVs with different CSI Drivers, but I can also add test-cases for other volumes kinds. Good point.